derivative/dglobal.py

```diff
@@ -7,40 +7,47 @@
 from scipy import interpolate, sparse
 from scipy.special import legendre
 from sklearn.linear_model import Lasso
+from specderiv import cheb_deriv, fourier_deriv
 
 
 @register("spectral")
```

Comment:
Is the `methods` dictionary used for anything other than testing? It's kind of nice to have a list of all the methods, but the way everything is tested together with the `compare` method with super loose bounds of `1e-1` strikes me as not very ideal. Seems like an example of abstraction run amok to me. And then you end up with special cases where you have to say `bad_combo=True` or create `f_mod`. We might be better served by testing each thing individually and setting tighter bounds on each one's performance.

It's on my mind because I'm trying to figure out how to include tests for both the Fourier and Chebyshev cases, and as written I can only test one set of `kwargs`. I can put a loop wherever to cover both, of course, but that's more special-case junk in the test code.

Reply:
Basically, we're only testing some particular set of params to each of these methods right now, and the way the code is written dissuades more thorough testing. That can't be good for true coverage.

Reply:
General guidance on PRs: if your method meets or improves the standard of the codebase, its good enough.

I have no idea what order of accuracy each method should approach